numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -570,24 +595,37 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
 static int clipmode_parser(char const *str, Py_ssize_t length, void *data)
 {
     NPY_CLIPMODE *val = (NPY_CLIPMODE *)data;
+    int is_exact = 0;
+
     if (length < 1) {
         return -1;
     }
+
     if (str[0] == 'C' || str[0] == 'c') {
         *val = NPY_CLIP;
-        return 0;
+        is_exact = (length == 4 && strcmp(str, "clip") == 0);
     }
     else if (str[0] == 'W' || str[0] == 'w') {
         *val = NPY_WRAP;
-        return 0;
+        is_exact = (length == 4 && strcmp(str, "wrap") == 0);
     }
     else if (str[0] == 'R' || str[0] == 'r') {
         *val = NPY_RAISE;
-        return 0;
+        is_exact = (length == 5 && strcmp(str, "raise") == 0);
     }
-    else {
```

Comment:
I don't think you want to remove this

Reply:
aah! i think i should probably add a test for this use case.